homeassistant/components/heos/config_flow.py

```diff
@@ -33,12 +33,17 @@ async def async_step_ssdp(self, discovery_info):
         # Abort if other flows in progress or an entry already exists
         if self._async_in_progress() or self._async_current_entries():
             return self.async_abort(reason="already_setup")
+        await self.async_set_unique_id(DOMAIN)
         # Show selection form
         return self.async_show_form(step_id="user")
 
     async def async_step_import(self, user_input=None):
         """Occurs when an entry is setup through config."""
         host = user_input[CONF_HOST]
+        # raise_on_progress is False here in case ssdp discovers
+        # heos first which would block the import
+        await self.async_set_unique_id(DOMAIN, raise_on_progress=False)
+        self._abort_if_unique_id_configured()
```

Comment:
This blocks import the same way as `async_set_unique_id(DOMAIN, raise_on_progress=True)` would.  Remove this line as the import from `configuration.yaml` should take precedence over discovery.

I don't think there's any way to prevent the discovery notification from showing if discovery detects HEOS before it's imported from config for the first time--which shouldn't happen.  Subsequent restarts won't have this problem.